lib/install/management/appliance.go

```diff
@@ -120,25 +121,50 @@ func (d *Dispatcher) checkExistence(conf *config.VirtualContainerHostConfigSpec,
 		return err
 	}
 	if orp == nil {
-		return nil
+		return nil // TODO[zjs]: Check for existing VM Group in this case too
 	}
 
 	rp := compute.NewResourcePool(d.op, d.session, orp.Reference())
 	vm, err := rp.GetChildVM(d.op, d.session, conf.Name)
 	if err != nil {
 		return err
 	}
-	if vm == nil {
-		return nil
+	if vm != nil {
+		d.op.Debug("Appliance is found")
+		if ok, verr := d.isVCH(vm); !ok {
+			verr = errors.Errorf("Found virtual machine %q, but it is not a VCH. Please choose a different virtual app.", conf.Name)
+			return verr
+		}
+		return errors.Errorf("Virtual app %q already exists. Please delete it before reinstalling.", conf.Name)
 	}
 
-	d.op.Debug("Appliance is found")
-	if ok, verr := d.isVCH(vm); !ok {
-		verr = errors.Errorf("Found virtual machine %q, but it is not a VCH. Please choose a different virtual app.", conf.Name)
-		return verr
+	if conf.UseVMGroup {
```

Comment:
`&& d.session.IsVC()`? Error if not VC? Or does the cluster code work on ESX and just ends as a no-op.

Reply:
- [ ] Test this directly against ESX and make any necessary changes.